app/models/query.rb

```diff
@@ -325,8 +325,11 @@ def sort_criteria_columns
         column = sortable_columns
                  .detect { |candidate| candidate.name == attribute }
 
-        [column, direction]
+        if column
+          [column, direction]
+        end
       end
+      .compact
```

Comment:
Removing invalid columns is inconsistent to the way filters are treated and will result in clients have less feedback and left wondering why a sorting is not applied when they e.g. have a spelling error. Any chance we can provide adequate feedback?

Reply:
This should already happen in all regular queries going through the form. This is only happening when directly acessing the `Query::Results` which happens only internally as far as I can tell.

Reply:
But when the query is invalid it is designed to return no results at all in all the other places. E.g. when attempting to fetch work packages using an invalid filter values, no results will be returned at all. 

The graceful degradation will only take place for stored queries to take different permissions and the change in allowed values after a query has been created.

As `parent` used to be a valid sort column before, how about we explicitly check for it here and only remove that particular column. Using `compact` is too coarse IMO.

Reply:
In what way is it 'correct' behavior that you a get a columns object of `[nil, 'asc']` ? This is just plain wrong. I agree that we should disallow access to the results alltogether (by checking for valid? and returning an empty scope), but that does not make this change incorrect.

Reply:
I did not want to suggest such a value to be correct. And I did also not want to suggest that the change is incorrect as such. 

But it is too lenient in my opinion and does not match the way the rest of query is designed.

Let's restrict the removal of a faulty column to `parent` only in order to support backwards compatibility for now, but let's not just swallow faulty column names. So checking the validity or explicitly accepting graceful degradation should be the long term goal.